cmd/repl/repl.go

```diff
@@ -230,6 +239,66 @@ func makeSessionFromEnv() *pb.Session {
 		Submitter:        os.Getenv("SQLFLOW_submitter")}
 }
 
+func switchDatabase(db string, session *pb.Session) {
+	stream := sql.RunSQLProgram("USE "+db, "", session)
+	r := <-stream.ReadAll()
+	switch r.(type) {
+	case string:
+		session.DbConnStr = getDataSource(session.DbConnStr, db)
+		fmt.Println("Database changed to", db)
+		currentDB = db
+	}
+}
+
+func getDatabaseName(datasource string) string {
+	driver, other, e := database.ParseURL(datasource)
+	if e != nil {
+		log.Fatalf("unrecognized data source '%s'", datasource)
+	}
+	re := regexp.MustCompile(`[^/]*/(\w*).*`) // mysql and hive
+	switch driver {
+	case "maxcompute":
+		re = regexp.MustCompile(`[^/].*/api[?].*curr_project=(\w*).*`)
+	case "mysql":
+	case "hive":
+	default:
+		log.Fatalf("unknown database '%s' in data source'%s'", driver, datasource)
+	}
+	if group := re.FindStringSubmatch(other); group != nil {
+		return group[1]
+	}
+	return ""
+}
+
+func getDataSource(dataSource, db string) string {
```

Comment:
We need a comment to explain what the function does.

Reply:
Done.